src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -62,19 +62,18 @@ public static void generate(String... args) throws IOException, CheckstyleExcept
         defaultConfiguration.addChild(treeWalkerConfig);
         treeWalkerConfig.addChild(scraperCheckConfig);
         checker.configure(defaultConfiguration);
-        dumpMetadata(checker, args[0]);
+        final List<File> validFiles = extractValidFiles(args[0]);
+        checker.process(validFiles);
```

Comment:
https://github.com/checkstyle/checkstyle/issues/10747#issue-981927301
> I also feel MetadataGeneratorUtil should report any errors back somehow so the client and calling tests can handle the situation appropriately.

Right now few things are looking at the result since the method doesn't specifically return it. We should be returning a result now as part of `MetadataGeneratorUtilTest` as we want users to call this method to generation and error reporting.